jax/_src/interpreters/partial_eval.py

```diff
@@ -1883,10 +1889,16 @@ def vars_in_shape(aval: AbstractValue) -> Sequence[Var]:
   new_invars = [var(v) for v in jaxpr.invars]
   new_eqns = []
   for eqn in jaxpr.eqns:
-    invars = [lit(v) or var(v) for v in eqn.invars]
+    invars = [lit(v) or var(v)
```

Comment:
Maybe cleaner to change the type signature of `lit` here rather than add this `isinstance` logic?

Same below.

Reply:
Good thinking.  I like your approach better